onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -182,10 +178,8 @@ template <typename T>
 OrtStatus* CreateTensorImpl(const int64_t* shape, size_t shape_len, const OrtAllocatorInfo* info,
                             void* p_data, size_t p_data_len, std::unique_ptr<Tensor>* out) {
   size_t elem_count = 1;
-  std::vector<int64_t> shapes(shape_len);
   for (size_t i = 0; i != shape_len; ++i) {
```

Comment:
Given TensorShape has code to do this can we create a TensorShape instance here, get Size() from it, and then std::move it when calling the Tensor ctor?

Should we also have a check somewhere to ensure there's no symbolic dimension with value of -1 leading to a negative total size?

Reply:
I wrote the code like this because, anyway I need to loop the shape array. If call Size() from TensorShape, it means the shape array need be looped twice.  One for copy the data into TensorShape, another for calculating the Size().  So I tried to combine these two into one loop.

Reply:
I will figure out this in a separated PR.